packages/react-native/ReactCommon/react/renderer/core/RawValue.h

```diff
@@ -275,6 +510,33 @@ class RawValue {
     }
     return result;
   }
+
+  template <typename T>
+  static std::unordered_map<std::string, T> castValue(
+      const jsi::Value& value,
+      jsi::Runtime* runtime,
+      std::unordered_map<std::string, T>* /*type*/) {
+    react_native_assert(value.isObject());
+    jsi::Object object = value.asObject(*runtime);
+    auto propertyNames = object.getPropertyNames(*runtime);
+    auto size = propertyNames.size(*runtime);
+    auto result = std::unordered_map<std::string, T>{};
+    for (size_t i = 0; i < size; i++) {
+      jsi::Value propertyNameValue = propertyNames.getValueAtIndex(*runtime, i);
+      jsi::String propertyName = propertyNameValue.getString(*runtime);
+      jsi::Value propertyValue = object.getProperty(*runtime, propertyName);
+      if (propertyValue.isUndefined()) {
+        // Skip undefined values to mimic JSIDynamic::dynamicFromValue behavior.
+        // Null values are allowed in the map.
+        continue;
+      }
+
+      std::string propertyNameString = propertyName.utf8(*runtime);
+      T property = castValue(propertyValue, runtime, (T*)nullptr);
+      result[propertyNameString] = property;
```

Comment:
Use `.emplace(` and move `property` into it for better performance. No need for a copy here.